pkg/provisioner/ironic/ironic.go

```diff
@@ -352,7 +353,10 @@ func (p *ironicProvisioner) ValidateManagementAccess(data provisioner.Management
 		driverInfo["deploy_iso"] = p.config.deployISOURL
 	}
 
-	// If we have not found a node yet, we need to create one
+	if data.CurrentImage != nil && data.CurrentImage.IsLiveISO() && p.config.liveISOForcePersistentBootDevice != "" {
+		driverInfo["force_persistent_boot_device"] = p.config.liveISOForcePersistentBootDevice
+	}
```

Comment:
I'm not sure there's any need to set this here, given that we do getImageUpdateOptsForNode() later anyway.

We're not even always saving a potential write in the case where we're creating a new Node, because if liveISOForcePersistentBootDevice is set but the current image is not a live ISO, we're not doing anything here but we will set driver_info/force_persistent_boot_device to `"Default"` later.

Reply:
Ack yeah I wasn't entirely sure if we needed it in both places, if we're sure this will work OK on the adopt case if we only set it in getImageUpdateOptsForNode() then removing it sounds fine - I'm assuming Ironic only asserts this boot-device during provisioning @derekhiggins?

Reply:
Ya, ironic will only verify BMC on adopt. I'll push up a new version without these lines.